packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -521,11 +521,16 @@ export function getEnvForPackageManager({
   lockfileVersion,
   nodeVersion,
   env,
+  showLogs = true,
 }: {
   cliType: CliType;
   lockfileVersion: number | undefined;
   nodeVersion: NodeVersion | undefined;
   env: { [x: string]: string | undefined };
+  /**
+   * Whether to print out on console what lockfile was detected.
+   */
+  showLogs?: boolean | undefined;
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** change side effects and return type

I think this would work better if the `getEnvForPackageManager` returned the new `PATH` and `YARN_NODE_LINKER` values, then the consumer would slap those onto the current env vars and log whatever it wants.

If you don't want to change the signature and side effects of functions that cross the package boundary, I suggest extracting the logic that determines the `PATH` and `YARN_NODE_LINKER` values into a new function that is also exported, allowing both options for consumer code.‏